packages/helpers/src/modules/assetValue.ts

```diff
@@ -336,17 +371,70 @@ function getAssetInfo(identifier: string) {
     isSynthetic,
     isSynthOrTrade,
     isTradeAsset,
-    symbol,
+    symbol: finalSymbol,
     ticker,
   };
 }
 
+/**
+ * Ticker is everything before the last hyphen. Address is after.
+ * This is the standard for EVM chains and most others.
+ * Example: "PENDLE-LPT-0x123..." -> { ticker: "PENDLE-LPT", address: "0x123..." }
+ */
+function parseWithLastSeparator(symbol: string) {
+  const lastDashIndex = symbol.lastIndexOf("-");
+  if (lastDashIndex === -1) {
+    return { address: undefined, ticker: symbol };
+  }
+  const ticker = symbol.substring(0, lastDashIndex);
+  const address = symbol.substring(lastDashIndex + 1);
+  return { address, ticker };
+}
+
+/**
+ * Ticker is everything before the first hyphen. Address is after.
+ * This is specific to NEAR's convention.
+ * Example: "PURGE-purge.near" -> { ticker: "PURGE", address: "purge.near" }
+ */
+function parseWithFirstSeparator(symbol: string) {
+  const firstDashIndex = symbol.indexOf("-");
+  if (firstDashIndex === -1) {
+    return { address: undefined, ticker: symbol };
+  }
+  const ticker = symbol.substring(0, firstDashIndex);
+  const address = symbol.substring(firstDashIndex + 1);
+  return { address, ticker };
+}
```

Comment:
Those are "exactly` the same functions - the only difference is first line.


```suggestion
function parseWithSeparator({ symbol, type = "first" }: { symbol: string, type: "first" | "last" }) {
  const dashIndex = type === "first" ? symbol.indexOf("-") : symbol.lastIndexOf("-");
  
  if (dashIndex === -1) {
    return { address: undefined, ticker: symbol };
  }
  
  const ticker = symbol.substring(0, dashIndex);
  const address = symbol.substring(dashIndex + 1);
  return { address, ticker };
}